### PR TITLE
Enable ESLint

### PR DESCRIPTION
**Pull request by cjdenio**
This PR enables ESLint, and fixes all relevant errors (you'd be surprised how much wacky stuff it caught)

### CODE REVIEW

**Review by sampoder — COMMENTED** [review]
I'm not against the changes, but I'd like not to add eslintrc as it just become a pain in the neck in Next.js eg. https://hackclub.slack.com/archives/C019RMWTECD/p1626074720234300

**Comment by lachlanjc** [comment]
#### 📝 Changed routes:


| [**/bank**](https://v3-92peyd23f.hackclub.dev/bank) | [**/**](https://v3-92peyd23f.hackclub.dev/) |
|:-:|:-:|
|<a href="https://v3-92peyd23f.hackclub.dev/bank">Screenshot of /bank</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/8Pt5sNWf38cSUaqS60eeLAMU__p7DlXqYU1gCAkgPHv7Yd1KNyQVYgdFTCudgLijP81CVWC1gDlvyDzRRuAB9g.png">(view full size)</a> |<a href="https://v3-92peyd23f.hackclub.dev/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/-lZMccU4lxfSpNrpLvRIYfs_OengsWAiUm0iQ1i_t-49y7h-S7qEilHCV5XpTGACIl8NWkE1Fb0wRrRNYwHVKg.png">(view full size)</a> |



| [**/jobs/bank-ops-associate**](https://v3-92peyd23f.hackclub.dev/jobs/bank-ops-associate) | [**/jobs/bank-ops-lead**](https://v3-92peyd23f.hackclub.dev/jobs/bank-ops-lead) |
|:-:|:-:|
|<a href="https://v3-92peyd23f.hackclub.dev/jobs/bank-ops-associate">Screenshot of /jobs/bank-ops-associate</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/9ZBVaCU1VjRXGIKEghV1oT32BvNxyGRMJcLENhGXqvoLRNyJB4ayRcc5jRsCn6ma4eVag4UZgtTVen_qFVKZDw.png">(view full size)</a> |<a href="https://v3-92peyd23f.hackclub.dev/jobs/bank-ops-lead">Screenshot of /jobs/bank-ops-lead</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/dN4Dfsqh0zDRB4HrI_HUu9JF7dEE5toEck26SK_jgau2fNHSjKAMzALw5-Fgyy2U-Ug_z5VAu6Ph6BPGhrkpDQ.png">(view full size)</a> |



| [**/jobs/bank-tech-lead**](https://v3-92peyd23f.hackclub.dev/jobs/bank-tech-lead) | [**/jobs/brand-director**](https://v3-92peyd23f.hackclub.dev/jobs/brand-director) |
|:-:|:-:|
|<a href="https://v3-92peyd23f.hackclub.dev/jobs/bank-tech-lead">Screenshot of /jobs/bank-tech-lead</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/3ZGoDMmmsXOBO5yHXOEn4lna603M-CRzPESpgSVdCp1LbL1G0MeA_-6q-4eeDrTlD4XjvnJncKAlP47KrTgzig.png">(view full size)</a> |<a href="https://v3-92peyd23f.hackclub.dev/jobs/brand-director">Screenshot of /jobs/brand-director</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/qiO7bDnOWBCEbCl63I-5h7mQBQhHfsfUlAuT2hVz-G6x4OYpoyGVKqJHNB9jsTpuT6H1slDsay1R6j7E1vSJoA.png">(view full size)</a> |



And 17 other routes:
- [**/api/channels/dogs**](https://v3-92peyd23f.hackclub.dev/api/channels/dogs)
- [**/api/join**](https://v3-92peyd23f.hackclub.dev/api/join)
- [**/api/stickers**](https://v3-92peyd23f.hackclub.dev/api/stickers)
- [**/jobs/clubs-lead**](https://v3-92peyd23f.hackclub.dev/jobs/clubs-lead)
- [**/jobs/communications-manager**](https://v3-92peyd23f.hackclub.dev/jobs/communications-manager)
- [**/jobs/education-engineer**](https://v3-92peyd23f.hackclub.dev/jobs/education-engineer)
- [**/jobs/events-designer**](https://v3-92peyd23f.hackclub.dev/jobs/events-designer)
- [**/jobs/executive-assistant**](https://v3-92peyd23f.hackclub.dev/jobs/executive-assistant)
- [**/jobs**](https://v3-92peyd23f.hackclub.dev/jobs)
- [**/jobs/journalist**](https://v3-92peyd23f.hackclub.dev/jobs/journalist)
- [**/jobs/philanthropy-position**](https://v3-92peyd23f.hackclub.dev/jobs/philanthropy-position)
- [**/minecraft**](https://v3-92peyd23f.hackclub.dev/minecraft)
- [**/night**](https://v3-92peyd23f.hackclub.dev/night)
- [**/opensource**](https://v3-92peyd23f.hackclub.dev/opensource)
- [**/philosophy**](https://v3-92peyd23f.hackclub.dev/philosophy)
- [**/preston-werner**](https://v3-92peyd23f.hackclub.dev/preston-werner)
- [**/santa**](https://v3-92peyd23f.hackclub.dev/santa)



Commit c68fe8b5325dd4941ed03b667d10b3bae23362eb (https://v3-92peyd23f.hackclub.dev).

File: components/bank/Nonprofits.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ export default function Nonprofits() {
         <Heading variant="title">Nonprofit? No problem.</Heading>
         <Text variant="lead" color="muted">
           Hack Club Bank is a powerful, safe, and easy-to-use financial
-          platform, whether you're receiving your first donation or spend
+          platform, whether you&apos;re receiving your first donation or spend
```
Comment (sampoder) [inline components/bank/Nonprofits.js]: Curious, how come we need to do this? Personally I think it just makes the code less readable.

**Comment by cjdenio** [comment]
Rules can be disabled on a per-rule basis (disabling the `` -> `<Image>` thing is a great idea), but I think ESLint is wildly helpful in most cases. Again, it's crazy how many potentially hazardous things it found (like missing `key` props and such). Up to you, though.

**Comment by sampoder** [comment]
Is there a way to make it more friendly? Basically it lets you know but I doesn't block your build

File: components/bank/Nonprofits.js (hunk 1 above)
Comment (cjdenio) [reply components/bank/Nonprofits.js]: It basically prevents problems surrounding [misplaced angle brackets](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md), but yeah, I'll remove it

**Comment by cjdenio** [comment]
Yep! https://nextjs.org/docs/api-reference/next.config.js/ignoring-eslint

**Review by sampoder — APPROVED** [review]
I am converting to the dark side

**Comment by cjdenio** [comment]
thanks sam!

image

**Comment by exu3** [comment]
Screen Shot 2021-09-25 at 6 25 43 PM

**Pull request merged**